README.md

```diff
@@ -157,6 +157,106 @@ This is useful for long examples.
 
 [^back to top](#recog-a-recognition-framework)
 
+### Examples
+
+The following examples make use of `bin/recog_match`, a simple Ruby command line tool that uses Recog's fingerprinting data. Pre-processing is generally required before running Recog, i.e. extracting HTTP header values, etc. All fingerprint data can be found in `xml/*.xml`.
+
+### ftp_banners
+
+Fingerprint FTP servers based on the server's banner response after connecting:
+
+```
+# Example plaintext input
+echo -n '220 Microsoft FTP Service' | bin/recog_match xml/ftp_banners.xml -
+
+# Example command using nmap
+nmap -sV -script=banner -p 21 192.168.123.13 | grep '_banner' | cut -d: -f2- | bin/recog_match xml/ftp_banners.xml -
+
+# Example output
+MATCH: {"matched"=>"Microsoft FTP Server on Windows XP, 2003 or later without version", "service.vendor"=>"Microsoft", "service.product"=>"IIS", "service.family"=>"IIS", "service.cpe23"=>"cpe:/a:microsoft:iis:-", "os.vendor"=>"Microsoft", "os.family"=>"Windows", "os.product"=>"Windows", "os.cpe23"=>"cpe:/o:microsoft:windows:-", "host.name"=>"220", "service.protocol"=>"ftp", "fingerprint_db"=>"ftp.banner", "data"=>"220 Microsoft FTP Service"}
+```
+
+#### http_cookies
+
+Using the HTTP `Set-Cookie` header value to fingerprint an HTTP server:
+
+```
+# Example plaintext input
+echo 'laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax' | bin/recog_match xml/http_cookies.xml -
+
+# Example cURL command
+curl --silent -I http://localhost:9001 | grep -i '^Set-Cookie:' | cut -d: -f2- | bin/recog_match xml/http_cookies.xml -
+
+# Example output
+MATCH: {"matched"=>"Laravel PHP web application framework", "service.vendor"=>"Laravel", "service.product"=>"Laravel", "service.cpe23"=>"cpe:/a:laravel:laravel:-", "service.protocol"=>"http", "fingerprint_db"=>"http_header.cookie", "data"=>"laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax"}
+```
+
+#### http_servers
+
+Using the HTTP `Server` header value to fingerprint an HTTP server:
+
+```
+# Example cURL command
+curl --silent -I http://localhost:9001 | grep -i '^Server:' | cut -d: -f2- | bin/recog_match xml/http_servers.xml -
+
+# Example plaintext input
+echo -n 'Apache/2.4.38 (Debian)' | bin/recog_match xml/http_servers.xml -
```

Comment:
Move the `echo` example to the beginning for consistency with the other examples.